# Практическое задание 6

## по теме: "Взаимосвязь величин. Параметрические и непараметрические показатели корреляции. Корреляционный анализ".

**Задача 1.**

Даны значения величины заработной платы заемщиков банка (zp) и значения их поведенческого кредитного скоринга (ks):

zp = [35, 45, 190, 200, 40, 70, 54, 150, 120, 110],

ks = [401, 574, 874, 919, 459, 739, 653, 902, 746, 832].

Найдите ковариацию этих двух величин с помощью элементарных действий, а затем с помощью функции cov из numpy.
Полученные значения должны быть равны.

Найдите коэффициент корреляции Пирсона с помощью ковариации и среднеквадратичных отклонений двух признаков,
а затем с использованием функций из библиотек numpy и pandas.

In [1]:
import pandas as pd

In [2]:
zp = [35, 45, 190, 200, 40, 70, 54, 150, 120, 110]

In [3]:
ks = [401, 574, 874, 919, 459, 739, 653, 902, 746, 832]

In [4]:
zp_df = pd.DataFrame(zp)

In [5]:
ks_df = pd.DataFrame(ks)

$$cov_{XY} = M[(X - M(X))(Y - M(Y))] = M(XY) - M(X)M(Y)=\overline{X \cdot Y} - \overline{X} \cdot \overline{Y}$$

In [6]:
cov_0 = (zp_df * ks_df).mean() - zp_df.mean() * ks_df.mean()
cov_0

0    9157.84
dtype: float64

In [7]:
import numpy as np

In [8]:
cov_1 = np.cov(zp, ks)[0, 1] # Несмещенная оценка
cov_1

10175.377777777776

In [9]:
cov_0 = np.cov(zp, ks, ddof=0)[0, 1] # Смещенная оценка
cov_0

9157.84

$$r_{XY} = \frac{cov_{XY}}{\sigma_{X}\sigma_{Y}}$$

In [10]:
r = cov_1 / (zp_df.std(ddof=1) * ks_df.std(ddof=1))
r

0    0.88749
dtype: float64

**Задача 2.**

Измерены значения IQ выборки студентов,
обучающихся в местных технических вузах:
131, 125, 115, 122, 131, 115, 107, 99, 125, 111.
Известно, что в генеральной совокупности IQ распределен нормально.
Найдите доверительный интервал для математического ожидания с надежностью 0.95.

$$T_{1,2} = \overline{X} \pm \frac{s_0}{\sqrt{n}} \cdot c_{\gamma}$$

где $T_1$, $T_2$ — нижняя и верхняя границы доверительного интервала,

$\overline{X}$  — выборочное среднее арифметическое, $s_0$ — среднее квадратичное отклонение по выборке (несмещенное),

$n$ — размер выборки,

$\gamma$ — доверительная вероятность.

$c_{\gamma} = Ф^{-1} \frac{(1 + \gamma)}{2}$ — обратное значение функции стандартного нормального распределения.

In [11]:
IQ = [131, 125, 115, 122, 131, 115, 107, 99, 125, 111]

In [12]:
IQ_df = pd.DataFrame(IQ)

In [13]:
IQ_df.mean()

0    118.1
dtype: float64

In [14]:
s_0 = IQ_df.std(ddof=1)
s_0

0    10.545668
dtype: float64

t(alfa/2) = 2.262

In [15]:
n = len(IQ_df)
n

10

In [16]:
s_0 * 2.262 / n ** (0.5)

0    7.543392
dtype: float64

$$ [118.1 \pm 7.5] $$
$$ [110.6;  125.6] $$

**Задача 3.**

Известно, что рост футболистов в сборной распределен нормально
с дисперсией генеральной совокупности, равной 25 кв.см. Объем выборки равен 27,
среднее выборочное составляет 174.2. Найдите доверительный интервал для математического
ожидания с надежностью 0.95.

$$T_{1,2} = \overline{X} \pm \frac{s_0}{\sqrt{n}} \cdot c_{\gamma} = 174.2 \pm \frac{5}{\sqrt{27}} \cdot 1.96 = 174.2 \pm 1.9$$
$$[172.3; 176.1]$$